### DISCUSSION TITLE
Does rubocop fix the "Safe part" of offences?

### DISCUSSION THREAD

**Discussion by stdedos** [rubocop/rubocop#13278] [Q&A] [answered]
e.g. the https://docs.rubocop.org/rubocop/cops_style.html#stylestringconcatenation says

> This cop is unsafe in `aggressive` mode, as it cannot be guaranteed that the receiver is actually a string, which can result in a false positive.

But that could mean that, if a receiver could be determined to be a string, then the offence should be auto-correctable safely?

Like in 

```ruby
'Hello' + user.name
user.name + '!!'
Pathname.new('/') + 'test'
```

could become

```ruby
"Hello #{user.name}"
user.name + '!!'
Pathname.new('/') + 'test'
```

Even with the cop in `aggressive` mode

**Comment by Earlopain** [comment]
RuboCop at the moment is missing functionality to mark individual offenses of a cop as either safe or unsafe. There has been a recent attempt for something like that at https://github.com/rubocop/rubocop/pull/13166. I encourage you to make an issue about this to put more eyes on it, there are more cops where this would be applicable.

**Reply by stdedos** [reply, marked as answer]
https://github.com/rubocop/rubocop/issues/13280
